### ISSUE TITLE
Upgrade from 1.1.5-1 -> 1.1.7 cluster is not working correctly

### ISSUE THREAD

**Issue by vladimir-cirkovic** [arangodb/kube-arangodb#739]
Following this:
https://github.com/arangodb/kube-arangodb#upgrading-the-operator-using-helm

While upgrading kube-arangodb from 1.1.5-1 to 1.1.7 at the first look upgrade went smoothly, but looks like resources are not visible to operator anymore.
arangodb-operator    	arangodb 	1       	2021-06-17 13:16:17.864612473 +0200 CEST	deployed	kube-arangodb-1.1.7    	3.5.0      
arangodb-operator-crd	arangodb 	18      	2021-06-17 13:15:49.23419806 +0200 CEST 	deployed	kube-arangodb-crd-1.1.7	3.5.0  

While simulating missing agent with `kube delete ...`
pod get stuck with `Waiting for 1 more finalizers to be removed`

In case of force pod gets deleted but new one is not recreated.

Noticed operator logs:
`2021-06-17T11:20:19Z DBG Failed to get number of servers error="Get \"http://staging.arangodb.svc:8529/_admin/cluster/numberOfServers\": context deadline exceeded" component=deployment deployment=staging operator-id=b68g6`

arangomembers _(kubectl get arangomembers)_
are going back and forward in couple of seconds - not sure whether this is related.

_env versions {
  arangodb version:3.7.10
  kubernates version: v1.19.9-gke.1400
}_

Downgrade to 1.1.5-1 fixes the issue with hanging pod and make cluster pods visible to operator.

Do I missed some step while upgrading between those two versions ?

**Comment by ajanikow** [comment]
Hello!

Can you try with 1.1.8 or 1.1.9?

Also, can you share service `staging` which is failing? Looks like pods were not able to join service.

Best Regards,
Adam Janikowski.

**Comment by golonzovsky** [comment]
hey, we tried with all versions after `1.1.5-1` including `1.1.10` with same results. 

full deployment definition:
```
apiVersion: "database.arangodb.com/v1"
kind: "ArangoDeployment"
metadata:
  name: "staging"
spec:
  mode: Cluster
  image: arangodb/arangodb:3.7.10
  bootstrap:
    passwordSecretNames:
      root: arangodb-root-password-stg
  metrics:
    enabled: true
    image: arangodb/arangodb-exporter:0.1.7
  externalAccess:
    type: None
  tls:
    caSecretName: None
  dbservers:
    annotations: &annotations
      "prometheus.io/scrape": "true"
      "prometheus.io/path": "/metrics"
      "prometheus.io/port": "9101"
      "cluster-autoscaler.kubernetes.io/safe-to-evict": "true"
    nodeSelector: &selector
      example.ch/type: standard
    resources:
      limits:
        cpu: "5"
        memory: 30Gi
      requests:
        cpu: "3"
        memory: 30Gi
    volumeClaimTemplate:
      spec:
        accessModes: [ "ReadWriteOnce" ]
        storageClassName: pd-ssd
        resources:
          requests:
            storage: 100Gi
  coordinators:
    annotations: *annotations
    nodeSelector: *selector
    resources:
      limits:
        cpu: "2"
        memory: 10Gi
      requests:
        cpu: 300m
        memory: 3Gi
  agents:
    annotations: *annotations
    nodeSelector: *selector
    volumeClaimTemplate:
      spec:
        accessModes: [ "ReadWriteOnce" ]
        storageClassName: pd-ssd
        resources:
          requests:
            storage: 10Gi
---
kind: Service
apiVersion: v1
metadata:
  name: arangodb-cluster
  annotations:
    "cloud.google.com/load-balancer-type": "Internal"
spec:
  selector:
    app: arangodb
    arango_deployment: staging
    role: coordinator
  type: LoadBalancer
  ports:
    - protocol: TCP
      port: 8529
      targetPort: 8529
```

**Comment by golonzovsky** [comment]
fixed in [`1.2.1`](https://github.com/arangodb/kube-arangodb/releases/tag/1.2.1), thanks!

**Comment by vladimir-cirkovic** [comment]
fixed in 1.2.1, thanks!
